app/controllers/transcribe_controller.rb

```diff
@@ -123,113 +123,124 @@ def needs_review
   def save_transcription
     old_link_count = @page.page_article_links.where(text_type: 'transcription').count
 
-    unless params[:quality_sampling_id].blank?
-      @quality_sampling = QualitySampling.find(params[:quality_sampling_id])
-    end
+    @quality_sampling = QualitySampling.find(params[:quality_sampling_id]) if params[:quality_sampling_id].present?
 
     if @page.field_based
       @field_cells = request.params[:fields]
       table_cells = @page.process_fields(@field_cells)
     end
 
     @page.attributes = page_params unless page_params.empty?
-    #if page has been marked blank, call the mark_blank code 
-    unless params[:page]['needs_review'] == '1'
-      mark_page_blank(redirect: 'transcribe') or return
-    end
-    #check to see if the page needs to be marked as needing review
+
+    # if page has been marked blank, call the mark_blank code
+    mark_page_blank(redirect: 'transcribe') || return unless params[:page]['needs_review'] == '1'
+
+    # check to see if the page needs to be marked as needing review
     needs_review
-    if params['save'] || params['save_to_incomplete'] || params['save_to_needs_review'] || params['save_to_transcribed'] || params['approve_to_transcribed']
+
+    save_to_incomplete = params[:save_to_incomplete] || params[:done_to_incomplete]
+    save_to_needs_review = params[:save_to_needs_review] || params[:done_to_needs_review]
+    save_to_transcribed = params[:save_to_transcribed] || params[:done_to_transcribed]
+    approve_to_transcribed = params[:approve_to_transcribed]
+
```

Comment:
in #4070 I have introduced separating the `save_to` and `done_to` buttons so we know when to progress to the next page when clicked save or done.

Same applies in here, where having clear distinction between saving and clicking done can simplify things rather than introduce a new status `review_incomplete`

Reply:
I think that I like that differentiation.  Nice job on the renaming -- this is much clearer.